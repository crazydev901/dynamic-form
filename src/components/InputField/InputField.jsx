import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const InputField = ({
  label,
  placeholder,
  required,
  type,
  value,
  onChange,
  isError = false,
}) => {
  return (
    <div className="w-full md:w-1/2 px-3 mb-3 md:mb-0">
      <label className="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2">
        {label}
        {required && "*"}
      </label>
      <input
        className={classnames(
          "appearance-none block w-full bg-gray-100 text-gray-700 border rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white",
          {
            "border-red-500": isError,
            "border-gray-500 ": !isError,
          }
        )}
        id="grid-first-name"
        placeholder={placeholder}
        // required={required}
        type={type}
        value={value}
        onChange={(e) => onChange(e.target.value)}
        pattern={type === "tel" ? null : null}
      />
    </div>
  );
};

InputField.propTypes = {
  label: PropTypes.string,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  type: PropTypes.string,
  isError: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func,
};

export default InputField;
