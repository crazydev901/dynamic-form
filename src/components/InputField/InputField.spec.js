import React from "react";
import InputField from "./InputField";

import { render } from "@testing-library/react";

import "@testing-library/jest-dom";

describe("<InputField /> spec", () => {
	it("redners the component", () => {
		const container = render(
			<InputField
				label="Test"
				placeholder="test input"
				required
				type="text"
				value="test value"
				onChange={() => {}}
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
