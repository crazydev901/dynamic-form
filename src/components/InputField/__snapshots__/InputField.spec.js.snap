// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> spec redners the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full md:w-1/2 px-3 mb-3 md:mb-0"
      >
        <label
          class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
        >
          Test
          *
        </label>
        <input
          class="appearance-none block w-full bg-gray-100 text-gray-700 border rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white border-gray-500 "
          id="grid-first-name"
          placeholder="test input"
          type="text"
          value="test value"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full md:w-1/2 px-3 mb-3 md:mb-0"
    >
      <label
        class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
      >
        Test
        *
      </label>
      <input
        class="appearance-none block w-full bg-gray-100 text-gray-700 border rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white border-gray-500 "
        id="grid-first-name"
        placeholder="test input"
        type="text"
        value="test value"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
